File: study/ts/script/scene/view/team/TeamYokeConditionNode.ts
const { ccclass, property } = cc._decorator;

import CommonIconTemplate from '../../../ui/component/CommonIconTemplate'
import { Colors, G_UserData } from '../../../init';
import { Path } from '../../../utils/Path';
import CommonUI from '../../../ui/component/CommonUI';
import { handler } from '../../../utils/handler';
import { Lang } from '../../../lang/Lang';
import { UIPopupHelper } from '../../../utils/UIPopupHelper';
import PopupItemGuider from '../../../ui/PopupItemGuider';

@ccclass
export default class TeamYokeConditionNode extends cc.Widget {

    @property({
        type: cc.Node,
        visible: true
    })
    _resourceNode: cc.Node = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageBg: cc.Sprite = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageMark: cc.Sprite = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageMark2: cc.Sprite = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textName: cc.Label = null;

    @property({
        type: CommonIconTemplate,
        visible: true
    })
    _fileNodeIcon1: CommonIconTemplate = null;

    @property({
        type: CommonIconTemplate,
        visible: true
    })
    _fileNodeIcon2: CommonIconTemplate = null;

    @property({
        type: CommonIconTemplate,
        visible: true
    })
    _fileNodeIcon3: CommonIconTemplate = null;

    @property({
        type: CommonIconTemplate,
        visible: true
    })
    _fileNodeIcon4: CommonIconTemplate = null;

    @property({
        type: CommonIconTemplate,
        visible: true
    })
    _fileNodeIcon5: CommonIconTemplate = null;

    private _fateType:number;
    onLoad() {
        var size = this._resourceNode.getContentSize();
        this.node.setContentSize(size.width, size.height);
    }
    updateView(info) {
        this._textName.string = ('\u3010' + (info.name + '\u3011'));
        var color = info.isActivated && Colors.BRIGHT_BG_GREEN || Colors.BRIGHT_BG_TWO;
        this._textName.node.color = (color);
        var res = info.isActivated && Path.getUICommon('img_com_team_sign01') || Path.getUICommon('img_com_team_sign06');
        this._imageMark.addComponent(CommonUI).loadTexture(res);
        this._imageMark2.node.active = (info.isActivated);
        var heroIds = info.heroIds;
        this._fateType = info.fateType;
        for (var i = 1;i<=5;i++) {
            var heroId = heroIds[i-1];
            if (heroId) {
                (this['_fileNodeIcon' + i] as CommonIconTemplate).node.active = (true);
                (this['_fileNodeIcon' + i] as CommonIconTemplate).initUI(info.fateType, heroId);
                (this['_fileNodeIcon' + i] as CommonIconTemplate).setCallBack(handler(this, this._onClickIcon));
                (this['_fileNodeIcon' + i] as CommonIconTemplate).setTouchEnabled(true);
                if (this._isHave(info.fateType, heroId)) {
                    (this['_fileNodeIcon' + i] as CommonIconTemplate).setIconMask(false);
                } else {
                    (this['_fileNodeIcon' + i] as CommonIconTemplate).setIconMask(true);
                }
            } else {
                (this['_fileNodeIcon' + i] as CommonIconTemplate).node.active = (false);
            }
        }
    }
    _isHave(type, id) {
        if (type == 1) {
            return G_UserData.getTeam().isInBattleWithBaseId(id) || G_UserData.getTeam().isInReinforcementsWithBaseId(id);
        }else if (type ==2) {
            return G_UserData.getBattleResource().isEquipInBattleWithBaseId(id);
        } else if(type == 3) {
            return G_UserData.getBattleResource().isTreasureInBattleWithBaseId(id);
        }
        else if (type == 4) {
            return G_UserData.getBattleResource().isInstrumentInBattleWithBaseId(id);
        }
    }
    _onClickIcon(sender, itemParam) {
        UIPopupHelper.popupItemGuider(function(popupItemGuider:PopupItemGuider){
            popupItemGuider.updateUI(this._fateType, itemParam.cfg.id);
            popupItemGuider.setTitle(Lang.get('way_type_get'));
        }.bind(this))
    }
    onlyShow(info) {
        this._textName.string = ('\u3010' + (info.name + '\u3011'));
        var color = info.isActivated && Colors.BRIGHT_BG_GREEN || Colors.BRIGHT_BG_TWO;
        this._textName.node.color = (color);
        var res = info.isActivated && Path.getUICommon('img_sign01') || Path.getUICommon('img_sign01b');
        this._imageMark.addComponent(CommonUI).loadTexture(res);
        this._imageMark2.node.active = (info.isActivated);
        var heroIds = info.heroIds;
        for (var i = 1;i<=5;i++) {
            var heroId = heroIds[i-1];
            if (heroId) {
                (this['_fileNodeIcon' + i] as CommonIconTemplate).node.active = (true);
                (this['_fileNodeIcon' + i] as CommonIconTemplate).initUI(info.fateType, heroId);
                (this['_fileNodeIcon' + i] as CommonIconTemplate).setIconMask(!info.isActivated);
            } else {
                (this['_fileNodeIcon' + i] as CommonIconTemplate).node.active = (false);
            }
        }
    }

}

File: study/lua/src/app/scene/view/team/TeamYokeConditionNode.lua
--
-- Author: Liangxu
-- Date: 2017-03-29 18:13:54
-- 武将羁绊条件模块
local TeamYokeConditionNode = class("TeamYokeConditionNode", ccui.Widget)
local CSHelper = require("yoka.utils.CSHelper")
local TypeConvertHelper = require("app.utils.TypeConvertHelper")

function TeamYokeConditionNode:ctor()
	local resource = {
		file = Path.getCSB("TeamYokeConditionNode", "team"),
		binding = {

		}
	}
	CSHelper.createResourceNode(self, resource)

	self:_onCreate()
end

function TeamYokeConditionNode:_onCreate()
	local size = self._resourceNode:getContentSize()
	self:setContentSize(size.width, size.height)
end

function TeamYokeConditionNode:updateView(info)
	self._textName:setString("【"..info.name.."】")
	local color = info.isActivated and Colors.BRIGHT_BG_GREEN or Colors.BRIGHT_BG_TWO
	self._textName:setColor(color)
	local res = info.isActivated and Path.getUICommon("img_com_team_sign01") or Path.getUICommon("img_com_team_sign06")
	self._imageMark:loadTexture(res)
	self._imageMark2:setVisible(info.isActivated)

	local heroIds = info.heroIds
	self._fateType = info.fateType
	for i = 1, 5 do
		local heroId = heroIds[i]
		if heroId then
			self["_fileNodeIcon"..i]:setVisible(true)
			self["_fileNodeIcon"..i]:initUI(info.fateType, heroId)
			self["_fileNodeIcon"..i]:setCallBack(handler(self, self._onClickIcon))
			self["_fileNodeIcon"..i]:setTouchEnabled(true)
			if self:_isHave(info.fateType, heroId) then
				self["_fileNodeIcon"..i]:setIconMask(false)
			else
				self["_fileNodeIcon"..i]:setIconMask(true)
			end
		else
			self["_fileNodeIcon"..i]:setVisible(false)
		end
	end
end

function TeamYokeConditionNode:_isHave(type, id)
	if type == 1 then --武将羁绊
		return G_UserData:getTeam():isInBattleWithBaseId(id) or G_UserData:getTeam():isInReinforcementsWithBaseId(id)
	elseif type == 2 then --装备羁绊
		return G_UserData:getBattleResource():isEquipInBattleWithBaseId(id)
	elseif type == 3 then --宝物羁绊
		return G_UserData:getBattleResource():isTreasureInBattleWithBaseId(id)
	elseif type == 4 then --神兵羁绊
		return G_UserData:getBattleResource():isInstrumentInBattleWithBaseId(id)
	end
end

function TeamYokeConditionNode:_onClickIcon(sender, itemParam)
	local PopupItemGuider = require("app.ui.PopupItemGuider").new(Lang.get("way_type_get"))
    PopupItemGuider:updateUI(self._fateType, itemParam.cfg.id)
    PopupItemGuider:openWithAction()
end

function TeamYokeConditionNode:onlyShow(info)
	self._textName:setString("【"..info.name.."】")
	local color = info.isActivated and Colors.BRIGHT_BG_GREEN or Colors.BRIGHT_BG_TWO
	self._textName:setColor(color)
	local res = info.isActivated and Path.getUICommon("img_sign01") or Path.getUICommon("img_sign01b")
	self._imageMark:loadTexture(res)
	self._imageMark2:setVisible(info.isActivated)

	local heroIds = info.heroIds
	for i = 1, 5 do
		local heroId = heroIds[i]
		if heroId then
			self["_fileNodeIcon"..i]:setVisible(true)
			self["_fileNodeIcon"..i]:initUI(info.fateType, heroId)
			self["_fileNodeIcon"..i]:setIconMask(not info.isActivated)
		else
			self["_fileNodeIcon"..i]:setVisible(false)
		end
	end
end

return TeamYokeConditionNode